pkg/pillar/cmd/nim/nim.go

```diff
@@ -791,7 +790,26 @@ func (n *nim) ingestDevicePortConfigFile(oldDirname string, newDirname string, n
 		return
 	}
 	dpc.DoSanitize(n.Log, true, false, "", true, true)
-	dpc.OriginFile = filename
+	// Compare file timestamp first against /run/zedagent/DevicePortConfig/zedagent.json
+	// (if it exists and is newer we ignore the file from /config), and then
+	// /persist/status/nim/DevicePortConfigList/global.json
+	// and . If the latter exists and is newer
+	// we ignore the /config file.
+	cmpFilename1 := "/run/zedagent/DevicePortConfig/zedagent.json"
+	cmpFilename2 := "/persist/status/nim/DevicePortConfigList/global.json"
+	configTime := fileutils.GetModTime(filename)
+	controllerTime1 := fileutils.GetModTime(cmpFilename1)
+	controllerTime2 := fileutils.GetModTime(cmpFilename2)
+	if configTime.Before(controllerTime1) {
+		n.Log.Noticef("Ignoring %s over %s since older. %v vs %v",
+			filename, cmpFilename1, configTime, controllerTime1)
+		return
+	}
+	if controllerTime1.IsZero() && configTime.Before(controllerTime2) {
+		n.Log.Noticef("Ignoring %s over %s since older. %v vs %v",
+			filename, cmpFilename2, configTime, controllerTime2)
+		return
+	}
```

Comment:
In general this is OK, but I wonder if it is really needed for nim because we are already checking for duplicity when a new (or old) DPC is  being added to DPCL: https://github.com/lf-edge/eve/blob/6f877a4bdf998de555fef2d6b38185efa20e2dce/pkg/pillar/dpcmanager/dpc.go#L67-L139
But maybe the linked function will need some reinforcements and UTs to ensure that it can handle this case properly.

Reply:
Note that I have changed this to rely on the sha256 of the file content instead of the timestamp. See update diffs.